src/instrumentation/podmutator_test.go

```diff
@@ -2,881 +2,1113 @@ package instrumentation
+			},
+			expectedErrStr: "multiple New Relic Instrumentation instances available, cannot determine which one to select",
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			langInsts, err := GetLanguageInstrumentations(test.instrumentations)
+			errStr := ""
+			if err != nil {
+				errStr = err.Error()
+			}
+			if test.expectedErrStr != errStr {
+				t.Errorf("unexpected error string. expected: %s, got: %s", test.expectedErrStr, errStr)
+			}
+			if diff := cmp.Diff(test.expectedLangInsts, langInsts); diff != "" {
+				t.Errorf("Unexpected diff (-want +got): %s", diff)
+			}
+		})
+	}
+}
+
+func TestGetSecretNameFromInstrumentations(t *testing.T) {
+	tests := []struct {
+		name               string
+		instrumentations   []*v1alpha2.Instrumentation
+		expectedSecretName string
+		expectedErrStr     string
+	}{
+		{
+			name: "none",
+		},
+		{
+			name:               "one, default",
+			instrumentations:   []*v1alpha2.Instrumentation{{Spec: v1alpha2.InstrumentationSpec{LicenseKeySecret: DefaultLicenseKeySecretName}}},
+			expectedSecretName: DefaultLicenseKeySecretName,
+		},
+		{
+			name:               "one, blank",
+			instrumentations:   []*v1alpha2.Instrumentation{{Spec: v1alpha2.InstrumentationSpec{}}},
+			expectedSecretName: DefaultLicenseKeySecretName,
+		},
+		{
+			name:               "one, other",
+			instrumentations:   []*v1alpha2.Instrumentation{{Spec: v1alpha2.InstrumentationSpec{LicenseKeySecret: "something-else"}}},
+			expectedSecretName: "something-else",
+		},
+		{
+			name: "two, one blank, the other the default",
+			instrumentations: []*v1alpha2.Instrumentation{
+				{Spec: v1alpha2.InstrumentationSpec{}},
+				{Spec: v1alpha2.InstrumentationSpec{LicenseKeySecret: DefaultLicenseKeySecretName}},
+			},
+			expectedSecretName: DefaultLicenseKeySecretName,
+		},
+		{
+			name: "three, one something else, one the default, one blank",
+			instrumentations: []*v1alpha2.Instrumentation{
+				{Spec: v1alpha2.InstrumentationSpec{LicenseKeySecret: "something-else"}},
+				{Spec: v1alpha2.InstrumentationSpec{LicenseKeySecret: DefaultLicenseKeySecretName}},
+				{Spec: v1alpha2.InstrumentationSpec{}},
+			},
+			expectedErrStr: "multiple key secrets",
+		},
+		{
+			name: "two, one blank, the other the something else",
+			instrumentations: []*v1alpha2.Instrumentation{
+				{Spec: v1alpha2.InstrumentationSpec{}},
+				{Spec: v1alpha2.InstrumentationSpec{LicenseKeySecret: "something-else"}},
+			},
+			expectedErrStr: "multiple key secrets",
+		},
+		{
+			name: "two, one the default, the other the something else",
+			instrumentations: []*v1alpha2.Instrumentation{
+				{Spec: v1alpha2.InstrumentationSpec{LicenseKeySecret: DefaultLicenseKeySecretName}},
+				{Spec: v1alpha2.InstrumentationSpec{LicenseKeySecret: "something-else"}},
+			},
+			expectedErrStr: "multiple key secrets",
+		},
+		{
+			name: "two, one blank, the other the something else",
+			instrumentations: []*v1alpha2.Instrumentation{
```

Comment:
Did you mean to test blank with "something-else" but you are testing with `DefaultLicenseKeySecretName`

Reply:
updated the name